configs/projects/pe-installer-runtime-main.rb

```diff
@@ -0,0 +1,177 @@
+    proj.setting(:gcc_bindir, "#{proj.gcc_root}/bin")
+    proj.setting(:tools_root, "C:/tools/pl-build-tools")
+    proj.setting(:cppflags, "-I#{proj.tools_root}/include -I#{proj.gcc_root}/include -I#{proj.includedir}")
+    proj.setting(:cflags, "#{proj.cppflags}")
+    proj.setting(:ldflags, "-L#{proj.tools_root}/lib -L#{proj.gcc_root}/lib -L#{proj.libdir} -Wl,--nxcompat -Wl,--dynamicbase")
+    proj.setting(:cygwin, "nodosfilewarning winsymlinks:native")
+  end
+
+  if platform.is_macos?
+    # For OS X, we should optimize for an older architecture than Apple
+    # currently ships for; there's a lot of older xeon chips based on
+    # that architecture still in use throughout the Mac ecosystem.
+    # Additionally, OS X doesn't use RPATH for linking. We shouldn't
+    # define it or try to force it in the linker, because this might
+    # break gcc or clang if they try to use the RPATH values we forced.
+    proj.setting(:cppflags, "-I#{proj.includedir}")
+    proj.setting(:cflags, "-march=core2 -msse4 #{proj.cppflags}")
+    proj.setting(:ldflags, "-L#{proj.libdir} ")
+  end
+
+  # These flags are applied in addition to the defaults in configs/component/openssl.rb.
+  proj.setting(:openssl_extra_configure_flags, [
+    'no-dtls',
+    'no-dtls1',
+    'no-idea',
+    'no-seed',
+    'no-weak-ssl-ciphers',
+    '-DOPENSSL_NO_HEARTBEATS',
+  ])
+
+  # What to build?
+  # --------------
+  #
+  if platform.name =~ /^redhatfips-7-.*/
+    # Link against the system openssl instead of our vendored version.
+    # This is also used by components within this vanagon project (i.e. curl, ruby, ca-bundle)
+    proj.setting(:system_openssl, true)
+  else
+    proj.component "openssl-#{proj.openssl_version}"
+  end
+
+  # Ruby and deps
+  proj.component "runtime-pe-installer"
+  proj.component "puppet-ca-bundle"
+  proj.component "ruby-#{proj.ruby_version}"
+
+  # Puppet dependencies
+  proj.component 'rubygem-deep_merge'
+  proj.component 'rubygem-text'
+  proj.component 'rubygem-locale'
+  proj.component 'rubygem-gettext'
+  proj.component 'rubygem-fast_gettext'
+  proj.component 'rubygem-semantic_puppet'
+
+  # R10k dependencies
+  proj.component 'rubygem-gettext-setup'
+
+  # Core dependencies
+  proj.component 'rubygem-ffi'
+  proj.component 'rubygem-minitar'
+  proj.component 'rubygem-multi_json'
+
+  proj.setting(:rubygem_net_ssh_version, '5.2.0')
+  proj.component 'rubygem-net-ssh'
+
+  # net-ssh dependencies for el8's OpenSSH default key format
+  # since we do not need these for Windows (`puppet infra run` does not work for Windows platforms),
+  #   and building these can finicky, don't install for Windows
+  unless platform.is_windows?
+    proj.component 'rubygem-bcrypt_pbkdf'
+    proj.component 'rubygem-ed25519'
+  end
+
+  # Core Windows dependencies
+  proj.component 'rubygem-win32-dir'
+  proj.component 'rubygem-win32-process'
+  proj.component 'rubygem-win32-security'
+  proj.component 'rubygem-win32-service'
```

Comment:
Puppet 7 is not using these gems anymore and puppet-agent 7 is not shipping then anymore. Is pe-installer still using them? I was thinking maybe we can drop all usage of this since they are old and might not be compatible with ruby 2.7

Reply:
Thanks, good catch, I went ahead and removed those.